Sources/Observation.swift

```diff
@@ -95,6 +99,30 @@ extension Defaults {
 		public func invalidate() {
 			object?.removeObserver(self, forKeyPath: key, context: nil)
 			object = nil
+			lifetimeTie = nil
+		}
+
+		private struct LifetimeTie {
+			weak var object: AnyObject?
+		}
+
+		private var lifetimeTie: LifetimeTie?
+		private static var lifetimeTieAssociationKey = AssociatedObject<UserDefaultsKeyObservation>()
+
+		public func tieToLifetimeOf(_ weaklyHeldObject: AnyObject) -> DefaultsObservation {
+			UserDefaultsKeyObservation.lifetimeTieAssociationKey[weaklyHeldObject] = self
+			lifetimeTie = LifetimeTie(object: weaklyHeldObject)
+			return self
+		}
+
```

Comment:
Any way we could make the lifetime tying logic reusable? So we could use it for other things too, not just in this module.


Something like this:

```swift
private let bindLifetimeAssociatedObjectKey = ObjectAssociation<[AnyObject]>()

func bindLifetime(of object: AnyObject, to target: AnyObject) {
	var retainedObjects = bindLifetimeAssociatedObjectKey[target] ?? []
	retainedObjects.append(object)
	bindLifetimeAssociatedObjectKey[target] = retainedObjects
}
```

But `target` should be weakly held and maybe we should return an untie function from it.

And any thoughts of the use of `bind` vs `tie`?

We also need a way for it to cleanup, for example, in this case, we need to call `.invalidate()`. Maybe a `deinit` closure argument?